source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -0,0 +1,30 @@
+module agora.test.ValidatorRecurringEnrollment;
+
+import agora.test.Base;
+
+import core.thread.osthread;
+
+import std.stdio;
+
+unittest
+{
+    TestConf conf = { validator_cycle : 3, txs_to_nominate : 1, recurring_enrollment : true };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    auto txes = genesisSpendable().map!(txb => txb.sign()).array();
+
+    Height expected_height;
+    foreach (tx; txes)
+    {
+        node_1.putTransaction(tx);
+        Thread.sleep(1.seconds);
+        network.expectBlock(++expected_height);
+    }
```

Comment:
It would be good to add an assert here for the `expected_height` after the `foreach`.